packages/plugin-svelte/src/index.ts

```diff
@@ -0,0 +1,76 @@
+import path from 'path';
+import { logger } from '@rsbuild/shared';
+import type { RsbuildPlugin, RsbuildPluginAPI } from '@rsbuild/core';
+
+export type PluginSvelteOptions = {};
+
+export function pluginSvelte(
+  options: PluginSvelteOptions = {},
+): RsbuildPlugin<RsbuildPluginAPI> {
+  return {
+    name: 'plugin-svelte',
+
+    setup(api) {
+      let sveltePath: string = '';
+      try {
+        // Resolve `svelte` package path from the project directory
+        sveltePath = path.dirname(
+          require.resolve('svelte/package.json', {
+            paths: [api.context.rootPath],
+          }),
+        );
+      } catch (err) {
+        logger.error(
+          `Cannot resolve \`svelte\` package under the project directory, did you forget to install it?`,
+        );
+        throw new Error(`Cannot resolve \`svelte\` package`, {
+          cause: err,
+        });
+      }
+
+      api.modifyBundlerChain(async (chain, { CHAIN_ID, isProd }) => {
+        const { default: sveltePreprocess } = await import('svelte-preprocess');
+
+        const rsbuildConfig = api.getNormalizedConfig();
+
+        chain.resolve.alias
+          .set('svelte', path.join(sveltePath, 'src/runtime'))
+          .end()
+          .extensions.add('.svelte')
+          .end()
+          .mainFields.prepend('svelte')
+          .end()
+          .set('conditionNames', ['svelte', 'browser', 'import']);
```

Comment:
TODO: we can use `...` wildcard in next Rspack version

Related issue: https://github.com/web-infra-dev/rspack/issues/4432